cirq/ops/common_gates.py

```diff
@@ -446,12 +446,11 @@ def _measurement_key_(self):
         return self.key
 
     def _channel_(self):
-        size = 2 ** self.num_qubits()
-        zero = np.zeros((size, size))
-        zero[0][0] = 1.0
-        one = np.zeros((size, size))
-        one[-1][-1] = 1.0
-        return (zero, one)
+        def delta(i):
+            result = np.zeros((2 ** self.num_qubits(), ) * 2)
+            result[i][i] = 1
+            return result
```

Comment:
1. This returns a generator instead of tuple, is that the intent?
2. Keeping the `size` allows to not repeat `2 ** self.num_qubits()` twice.

Reply:
oh good calls.  Yes made tuple.  pulled size outside.